test/validation/api/ipsec/ipsec.c

```diff
@@ -1076,8 +1118,13 @@ int ipsec_init(odp_instance_t *inst, odp_ipsec_op_mode_t mode)
 	}
 
 	odp_pool_param_init(&params);
-	params.pkt.seg_len = MAX_PKT_LEN;
-	params.pkt.len     = MAX_PKT_LEN;
+	if (suite_context.reass_ipv4) {
+		params.pkt.seg_len = MAX_FRAGS * MAX_PKT_LEN;
+		params.pkt.len     = MAX_FRAGS * MAX_PKT_LEN;
```

Comment:
Why is this needed?

Reply:
Is your question about both len & seg_len?

Reply:
Both.

I think the meaning of MAX_PKT_LEN should not be changed. If it still meant the maximum packet lenght, you would not need MAX_FRAGS * MAX_PKT_LEN multiplication here and in a few other places.

One bug in the code here is that the check against the max_len pkt pool capability is still done with MAX_PKT_LEN, so after this change the code could attempt to create a pool with a request for longer packets than supported.

Then I do not see why seg_len needs to be increased. Why does the segment length even matter?

Reply:
Well, we don't restrict how a reassembled packet is presented, right? Suppose we have 4 frags with 1500B each, then the "completely" reassembled packet would have size as 4*1500B. 

And if the implementation chooses to reassemble into one segment, then that would be having size as 4*1500B, right? I can fix the capabilities. 

Please comment if this aligns with your understanding.

Reply:
> And if the implementation chooses to reassemble into one segment

It may not be able to do it if we have configured the packet pool with smaller max segment size and the implementation has not increased the actually allocated segment sizes under the hood (which might not be so good).

One could think that if an implementation cannot reassemble fragments into a multi-segment packet, it probably should not advertise the capability in the first place. Perhaps that is too strict a viewpoint, as I suppose the API basically always allows an implementation to report incomplete reassembly.

I guess that for now we can just implement the validation tests in a way that expect reassembly to always work in the tested cases and relax the testing later if necessary. Or did you have a better idea (more reassembly capability fields or something)?

Reply:
> It may not be able to do it if we have configured the packet pool with smaller max segment size
May be I'll drop the seg_len change and will take up as a separate PR for both capability & adjustments in UT. Can I assume you agree to the change in pkt_len?

Reply:
> Can I assume you agree to the change in pkt_len?

No. As I commented, the code has the bug that it compares MAX_PKT_LEN against the pkt.max_len pool capability but then proceeds to create the pool with a larger max len value (i.e. MAX_PKT_LEN * MAX_NUM_FRAGS). To me it seems that all the MAX_PKT_LEN * MAX_NUM_FRAGS multiplications in the code are unnecessary and lead to fragility like this. MAX_PKT_LEN could simply be changed from 1500 to 1500 * MAX_NUM_FRAGS, leaving the other code as is.

Reply:
@JannePeltonen If you are fine with updating the macro, then I don't have any issues.